rllib/evaluation/postprocessing_v2.py

```diff
@@ -0,0 +1,220 @@
+import numpy as np
+import scipy
+import tree
+
+from typing import List
+
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.core.models.base import STATE_IN
+from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+from ray.rllib.policy.sample_batch import concat_samples, DEFAULT_POLICY_ID, SampleBatch
+from ray.rllib.utils.replay_buffers.episode_replay_buffer import _Episode
+from ray.rllib.utils.annotations import DeveloperAPI
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.typing import TensorType
+
+# TODO (simon): Make framework-agnostic.
+_, tf, _ = try_import_tf()
+
+
+@DeveloperAPI
+class Postprocessing:
+    """Constant definitions for postprocessing."""
+
+    ADVANTAGES = "advantages"
+    VALUE_TARGETS = "value_targets"
+
+
+@DeveloperAPI
+def postprocess_episodes_to_sample_batch(samples: List[List[_Episode]]) -> SampleBatch:
+    """Converts the result from sampling with `EnvRunner` to a `SampleBatch'.
+
+    Once the `SampleBatch` will be deprecated this function will be
+    deprecated, too.
+    """
+    batches = []
+    for sample in samples:
+        for episode in sample:
+            # If the episode is terminated or truncated it is
+            # ready to be converted to a `SampleBatch`
+            if episode.is_terminated or episode.is_truncated:
```

Comment:
Can we move this logic into `Episode.to_sample_batch()` itself? I think that's cleaner and more defensive for covering other situations.